common-authentication/src/main/java/org/alfresco/hxi_connector/common/adapters/auth/HxAuthenticationClient.java

```diff
@@ -91,8 +94,25 @@ public void configure()
     public AuthenticationResult authenticate(String tokenUri, ClientRegistration clientRegistration)
     {
         String body = createEncodedBody(clientRegistration);
-        int contentLength = body.getBytes(UTF_8).length;
+        log.atDebug().log("Authentication :: sending token request for {} client registration", clientRegistration.getRegistrationId());
+        return sendRequest(tokenUri, body);
+    }
+
+    @Override
+    public AuthenticationResult authenticate(String clientRegistrationId)
+    {
+        String body = createEncodedBody(clientRegistrationId);
+        OAuth2ClientProperties.Provider provider = oAuth2ClientProperties.getProvider().get(clientRegistrationId);
+        Objects.requireNonNull(provider, "Auth Provider not found for client registration id: " + clientRegistrationId);
```

Comment:
maybe you could use `EnsureUtils` (common module), `ensureNonNull()`? It will throw custom exception not NPE